pkg/i2gw/provider.go

```diff
@@ -32,6 +34,10 @@ import (
 // func at startup.
 var ProviderConstructorByName = map[ProviderName]ProviderConstructor{}
 
+// ProviderConfByName ProviderConstructorByName is a map of ProviderConf by a
+// provider name. Different Provider implementations should add their conf at startup.
+var ProviderConfByName = map[ProviderName]ProviderConf{}
```

Comment:
If these configurations will be static and provider-specific, why don't we initialize them as part of each `NewProvider` function?

Reply:
Yes, it makes total sense. Addressed in e902610